### PR TITLE
ChromeAi Rtd Provider: Initial Release

### PR DESCRIPTION
**Pull request by pm-azhar-mulla**
## Type of change

- [x] Feature

## Description of change

Reference: https://github.com/prebid/Prebid.js/issues/13054
This PR enhances the Real-Time Data (RTD) capabilities of Prebid.js by introducing the Chrome AI RTD Provider module, which now supports:
- Automatic page language detection via Chrome's Language Detection API
- Automatic content summarization/keyword generation via Chrome's Summarizer API

### Key Features:
**Language Detection**
- Detects the primary language of the visible page content using Chrome's AI LanguageDetector API
- Configurable confidence threshold (default: 0.8)
- Allows custom ORTB2 path for detected language
- Caches detected language per URL in localStorage

**Summarization/Keyword Generation**
- Uses Chrome's Summarizer API to extract contextual summaries or keywords from the page
- Configurable summary type (headline, tldr, teaser, etc.), format (markdown/plain-text), and length
- Output injected into OpenRTB bid requests via configurable ortb2 path

**Modular Architecture**
- Enable or disable each feature independently
- Built with extensibility in mind to support future Chrome AI APIs

### Implementation Details:
- Uses Chrome's self-accessible LanguageDetector and Summarizer APIs
- Compatible with Prebid.js RTD submodule architecture
- Adds detected language to ortb2Fragments.global.site.content.language (path is configurable)
- Similarly summary will be added to ortb2Fragments.global.site.content.keywords (path is configurable)
- Respects existing values in ortb2
- Includes comprehensive documentation and example implementation

### Testing:
- Unit tests included covering edge cases and configuration options
- Example page (integrationExamples/chromeai/japanese.html) to demonstrate functionality

Documentation: 
https://developer.chrome.com/docs/ai/language-detection
https://developer.chrome.com/docs/ai/summarizer-api

### CODE REVIEW

File: modules/.submodules.json [hunk 1]
```diff
@@ -126,7 +126,8 @@
       "symitriDapRtdProvider",
       "timeoutRtdProvider",
       "weboramaRtdProvider",
-      "wurflRtdProvider"
+      "wurflRtdProvider",
+      "chromeAiRtdProvider"
```
Comment (patmmccann) [inline modules/.submodules.json]: these have an order :)

File: modules/chromeAiRtdProvider.js [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+      confidence: 0.8,
+      ortb2Path: 'site.content.language' // Default path for language
+    },
+    summarizer: {
+      enabled: false,
+      type: 'headline', // 'headline' or 'paragraph'
+      format: 'markdown', // 'markdown' or 'plaintext'
+      length: 'short', // 'short', 'medium', or 'long'
+      ortb2Path: 'site.content.keywords', // Default path for keywords
+      cacheInLocalStorage: true // Whether to cache detected keywords in localStorage
```
Comment (patmmccann) [inline modules/chromeAiRtdProvider.js:28]: wouldnt the summary change from pageview to pageview? I am not sure I follow the utility of the cache

Comment (pm-azhar-mulla) [reply modules/chromeAiRtdProvider.js:28]: @patmmccann We're caching the summary based on the full URL, not just the domain. This means that different pages on a news site can have unique summaries. Since the API takes some time to generate the summary, caching the processed data ensures we don't miss the opportunity to include it in the bidRequest. By storing the result, we avoid redundant processing and improve overall efficiency.

File: modules/.submodules.json (hunk 1 above)
Comment (pm-azhar-mulla) [reply modules/.submodules.json]: Done

File: modules/chromeAiRtdProvider.js (hunk 2 above)
Comment (pm-azhar-mulla) [reply modules/chromeAiRtdProvider.js:28]: The data will be stored in localStorage in following format
chromeAi_detected_data:"{'http://example.com/chromeai/testPage.html':{'language':{'language':'la','confidence':0.8760119676589966}, 'keywords':['AI News: research, startups, regulations, government initiatives']}}"

Comment (patmmccann) [reply modules/chromeAiRtdProvider.js:28]: Makes sense ty!

File: modules/chromeAiRtdProvider.js [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+
+  if (activeInitializations.length === 0) {
+    logMessage(`${CONSTANTS.LOG_PRE_FIX} No features enabled for initialization.`);
+    return true; // Module is considered initialized if no features are active/enabled.
+  }
+
+  // Wait for all enabled features to attempt initialization
+  try {
+    const results = await Promise.all(activeInitializations);
+    // Consider init successful if at least one feature init succeeded, or if no features were meant to run.
```
Comment (will-yobi) [inline modules/chromeAiRtdProvider.js:345]: why do we want to return true if no features run?

File: modules/chromeAiRtdProvider.md [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+### Advanced Configuration
+
+Configure language detection and summarization with additional options:
+
+```javascript
+pbjs.setConfig({
+  realTimeData: {
+    dataProviders: [{
+      name: 'chromeAi',
+      waitForIt: true, // Set to true if auction should wait for both enabled features
```
Comment (will-yobi) [inline modules/chromeAiRtdProvider.md:51]: How long would it delay auction for model inference? Doesn't this depend on the user's hardware?

https://developer.chrome.com/docs/ai/language-detection

Comment (pm-azhar-mulla) [reply modules/chromeAiRtdProvider.md:51]: @will-yobi So it depends upon what value is provided by publisher in auctionDelay. As specified in the [document](https://docs.prebid.org/dev-docs/add-rtd-submodule.html#:~:text=auctionDelay%20or%20waitForIt), the above snippet is just an example, publisher can decide whether to mark it as true or false.
And yes, the processing will depend upon user's hardware, hence we are storing the results in localStorage, so that we can only process it once and use the value in subsequent auctions.

File: modules/chromeAiRtdProvider.js (hunk 3 above)
Comment (pm-azhar-mulla) [reply modules/chromeAiRtdProvider.js:345]: @will-yobi as mentioned in the comment, the module is initialized as per the config hence we are returning true. i don't think, it will make any difference if we return true or false. or Am i missing something here?

**Comment by patmmccann** [comment]
note if this merges after #13253 it will need a storage disclosure added
